### DISCUSSION TITLE
Source code drop Visual Studio performance

### DISCUSSION THREAD

**Discussion by mrpavlitzky** [OrchardCMS/OrchardCore#11964] [Q&A]
Hello,

I'd very much like to dive into the source code of Orchard and contribute some minor changes that I feel might come in valuable for future users, but I have trouble with running it on a 16gb RAM machine. I'm not sure if it's an inherent limitation of my machine and the only reasonable way is to extend the available memory, or is there a way to optimize the work of VS, which as far as I know had historically some documented memory management issues.

This is the usage on 'idle'. Just having VS open, the application not running. 
![Memory usage on 'idle'](https://i.ibb.co/nDCvXZ6/before-debugging.png)

Here's the usage after build, with the debugger running. As you can see it eats up enough memory to virtually unable me to run for example a web browser with multiple tabs along with it.
![Memory usage while debugging](https://i.ibb.co/1qrSZ1q/debugging.png)

VS runs out of memory basically everytime I use anything else on my computer simultaneously.

I'd be grateful for any suggestions, or at least a confirmation that anyone managed to work with the source code successfully running it on a 16gb RAM machine.

Thanks
YU

**Comment by ns8482e** [comment]
Use VS Code

**Comment by hishamco** [comment]
@yuriko-ushikawa I'm suffering from this too ;) what I almost did is unload all the projects that I'm not working (Change something) on, from configuration manager, this of course will reduce the memory usage at least and accelarate the build compliation


Otherwise you can use VS Code as @ns8482e mentioned

**Reply by ns8482e** [reply]
IMO VS2022 is worst release in VS history very poor performance
